src/bio/bio_buffer.c

```diff
@@ -637,8 +637,18 @@ rw_completion(void *cb_arg, int err)
 	D_ASSERT(xs_ctxt->bxc_blob_rw > 0);
 	xs_ctxt->bxc_blob_rw--;
 
-	if (biod->bd_result == 0 && err != 0) {
-		biod->bd_result = daos_errno2der(-err);
+	/* Induce NVMe Read/Write Error*/
+	if (biod->bd_update)
+		err = DAOS_FAIL_CHECK(DAOS_NVME_WRITE_ERR) ? -DER_IO : err;
+	else
+		err = DAOS_FAIL_CHECK(DAOS_NVME_READ_ERR) ? -DER_IO : err;
```

Comment:
It should be "err = DAOS_FAIL_CHECK(XXX) ? -EIO : err;", then calls daos_errno2der() when propagate the error to bd_result, otherwise, the SPDK error code will be returned to client when real error happens.

Sorry, I overlooked this in first round review.

Reply:
Done.